src/dev/i18n/README.md

```diff
@@ -0,0 +1,175 @@
+# I18n Build Tools
+
+## Default messages extraction tool
+
+### Description
+
+The tool is used to extract default messages from all `*.{js, ts, jsx, tsx, html, handlebars, hbs, pug}` files in provided plugins directories to a JSON file.\
+The tool uses Babel to parse code and build an AST for each file or JS expression if whole file parsing is impossible. So only static parsing is available for messages extraction. It means that no variables, function calls or dynamic expressions can be used for ids, messages and description, only strings.
+
+### I18n examples and syntax restrictions
+
+**Global restrictions**:
+
+Values of `defaultMessage` and `description` properties must be string literals, template literals w/o expressions or string-only concatenation expressions. Identifiers and any other expressions are disallowed.\
+Value of `id` can be only a string literal.
```

Comment:
__nit:__ This sentence seems logically connected to the text above, maybe let's keep it on the same line. And to keep it inline with the previous sentence "The `id`"? And in all cases below too.